### PR TITLE
ToB L01

### PR DESCRIPTION
**Pull request by hensha256**

### CODE REVIEW

File: test/Sync.t.sol [hunk 1]
```diff
@@ -119,58 +157,63 @@ contract SyncTest is Test, Deployers, GasSnapshot {
...
+        actions[3] = Actions.ASSERT_DELTA_EQUALS;
+        params[3] = abi.encode(currency2, address(router), 0);
 
         router.executeActions(actions, params);
     }
 
     /// @notice When there is a balance, no delta should be applied.
-    function test_settle_balanceInPool_shouldNotApplyDelta() public noIsolate {
+    function test_settle_balanceInPool_shouldNotApplyDelta() public {
         uint256 currency0Balance = currency0.balanceOf(address(manager));
```
Comment (saucepoint) [inline test/Sync.t.sol:183]: minor nit for consistency with other tests, but we should assertGt 0 right

**Review by saucepoint — COMMENTED** [review]
looks good, also should we add `/// @dev PoolManager is not unlocked` to `IProtocolFees.collectProtocolFees`?

**Review by saucepoint — DISMISSED** [review]
will defer to @snreynolds for one last look

File: src/PoolManager.sol [hunk 2]
```diff
@@ -383,4 +383,9 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
     function _getPool(PoolId id) internal view override returns (Pool.State storage) {
         return _pools[id];
     }
+
+    /// @notice Implementation of the _isUnlocked function defined in ProtocolFees
+    function _isUnlocked() internal view override returns (bool) {
```
Comment (snreynolds) [inline src/PoolManager.sol:388]: also use this function in the modifier?

File: src/test/PoolSettleTest.sol [hunk 3]
```diff
@@ -3,18 +3,22 @@ pragma solidity ^0.8.24;
...
+    function settle(Currency syncCurrency) external payable {
+        manager.unlock(abi.encode(syncCurrency));
     }
 
-    function unlockCallback(bytes calldata) external returns (bytes memory) {
+    function unlockCallback(bytes calldata data) external returns (bytes memory) {
         require(msg.sender == address(manager));
 
+        Currency syncCurrency = abi.decode(data, (Currency));
+        manager.sync(syncCurrency);
```
Comment (snreynolds) [inline src/test/PoolSettleTest.sol:20]: you shouldnt need to call sync here if the currency we are settling is native?

File: test/utils/Deployers.sol [hunk 4]
```diff
@@ -87,15 +87,6 @@ contract Deployers {
     uint160 hookPermissionCount = 14;
     uint160 clearAllHookPermissionsMask = ~uint160(0) << (hookPermissionCount);
 
-    modifier noIsolate() {
```
Comment (snreynolds) [inline test/utils/Deployers.sol:90]: What was this from?

Comment (hensha256) [reply test/utils/Deployers.sol:90]: it was so the tests could call `sync` in a separate call than the actions router stuff. But now sync has to be in there to be in the lock so we dont need it

File: src/test/PoolSettleTest.sol (hunk 3 above)
Comment (hensha256) [reply src/test/PoolSettleTest.sol:20]: Its only used for 1 test, and its a test where the synced currency is _different_ than whats being settled. So just had to move that inside this contract. But arguably given this contract is only used for 1 test we should just use actions router

Comment (hensha256) [reply src/test/PoolSettleTest.sol]: ok i deleted PoolSettleTest and rewrote it with actions router

**Review by snreynolds — APPROVED** [review]

File: src/PoolManager.sol [hunk 5]
```diff
@@ -277,7 +277,7 @@ contract PoolManager is IPoolManager, ProtocolFees, NoDelegateCall, ERC6909Claim
     }
 
     /// @inheritdoc IPoolManager
-    function sync(Currency currency) external {
+    function sync(Currency currency) external onlyWhenUnlocked {
```
Comment (wjmelements) [inline src/PoolManager.sol:280]: what is the reason for this change?

Comment (wjmelements) [reply src/PoolManager.sol:280]: It's unnecessary for the manager to be unlocked to `sync` before and after this change.
